sources/kernel/syscalls.c

```diff
@@ -320,14 +340,51 @@ BrResult sys_ipc(BrIpcArgs *args)
     return result;
 }
 
-BrResult sys_irq(MAYBE_UNUSED BrIrqArgs *args)
+BrResult sys_irq(BrIrqArgs *args)
 {
     if (!(task_self()->caps & BR_CAP_IRQ))
     {
         return BR_BAD_CAPABILITY;
     }
 
-    return BR_NOT_IMPLEMENTED;
+    Task *task = nullptr;
+
+    if (args->task == BR_TASK_SELF)
+    {
+        task_ref(task_self());
+        task = task_self();
+    }
+    else
+    {
+        task = (Task *)domain_lookup(task_self()->domain, args->task, BR_OBJECT_TASK);
+    }
```

Comment:
Do we really need the ability to bind IRQ to tasks other than task_self() since we can already send IRQs across domains

Reply:
maybe if a pci device ask to the pci server to map an irq ?

Reply:
Then the PCI driver send the IRQObject to the task through IPC